pypiserver/_app.py

```diff
@@ -42,9 +42,12 @@ def protector(*args, **kwargs):
                     raise HTTPError(
                         401, headers={"WWW-Authenticate": 'Basic realm="pypi"'}
                     )
-                if not config.auther(*request.auth):
-                    raise HTTPError(403)
-            return method(*args, **kwargs)
+                if (
+                    config.auther(*request.auth)
+                    and config.authorizer(request.auth[0], request.auth[1], self.action)
```

Comment:
I've made the `authorizer` also take user name and password because in some implementations the username is a dummy value and the password really has all the info in it. For example, if the password were an access token of some sort, then the username could take on any value and would not be helpful for authorization purposes.